### ISSUE TITLE
weird backspace behaviour with 'softtabstop' + inline virtual text (formerly: "inlay hints with rust files")

### ISSUE THREAD

**Issue by ibldzn** [neovim/neovim#24075]
### Problem

it's kind of hard describing it with words, I hope the video is pretty clear about the problem, also sorry for the poor video quality, basically I can't delete anything _before_ the inlay hint virtual text

https://github.com/neovim/neovim/assets/51160226/c85f7fdd-aa01-4a40-8540-310bf25e4ab4

### Steps to reproduce

# Steps to reproduce using "nvim -u minimal_init.lua"

```lua
--- CHANGE THESE
local pattern = "rust"
local cmd = { "rust-analyzer" }
-- Add files/folders here that indicate the root of a project
local root_markers = { ".git", ".editorconfig", "Cargo.toml", "Cargo.lock" }
-- Change to table with settings if required
local settings = vim.empty_dict()

vim.api.nvim_create_autocmd("FileType", {
	pattern = pattern,
	callback = function(args)
		local match = vim.fs.find(root_markers, { path = args.file, upward = true })[1]
		local root_dir = match and vim.fn.fnamemodify(match, ":p:h") or nil
		vim.lsp.start({
			name = "rust_analyzer",
			cmd = cmd,
			root_dir = root_dir,
			settings = settings,
			on_attach = function(client, bufnr)
				vim.lsp.buf.inlay_hint(bufnr, true)
			end,
		})
	end,
})
```

<sup>Shamelessly stolen from #24068</sup>

### Expected behavior

I expect the characters behind the inlay hint virtual text to be deleted

### Neovim version (nvim -v)

NVIM v0.10.0-dev-541+g96b94f8d77

### Vim (not Nvim) behaves the same?

no

### Operating system/version

Arch Linux

### Terminal name/version

alacritty 0.12.1 (5fdfd47f)

### $TERM environment variable

alacritty

### Installation

neovim-git from Arch User Repository (AUR)

**Comment by jaredmcqueen** [comment]
@ibldzn off topic but how did you get the inlay text to be dim?  Like as if it was a comment?

**Comment by ibldzn** [comment]
> @ibldzn off topic but how did you get the inlay text to be dim?  Like as if it was a comment?

I have the `LspInlayHint` highlight group linked to `NonText` which I set to `#5b676a`

**Comment by SleepySwords** [comment]
This seems to be because of `softtabstop`, which is in rust.vim ftplugin file, interfering with inline virtual text rather than inlay hints.

**Comment by zeertzjq** [comment]
Related: https://github.com/vim/vim/issues/12669

**Comment by zeertzjq** [comment]
Fixed by #28005
